cyral/resource_cyral_repository_user_account.go

```diff
@@ -0,0 +1,596 @@
+				NewResourceData: func() ResourceData {
+					return &UserAccountResource{}
+				},
+				NewResponseData: func(_ *schema.ResourceData) ResponseData {
+					return &CreateUserAccountResponse{}
+				},
+			},
+			ReadRepositoryUserAccountConfig,
+		),
+		ReadContext: ReadResource(ReadRepositoryUserAccountConfig),
+		UpdateContext: UpdateResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryUserAccountUpdate",
+				HttpMethod: http.MethodPut,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf(
+						"https://%s/v1/repos/%s/userAccounts/%s",
+						c.ControlPlane,
+						d.Get("repository_id").(string),
+						d.Id(),
+					)
+				},
+				NewResourceData: func() ResourceData {
+					return &UserAccountResource{}
+				},
+			},
+			ReadRepositoryUserAccountConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryUserAccountDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf(
+						"https://%s/v1/repos/%s/userAccounts/%s",
+						c.ControlPlane,
+						d.Get("repository_id").(string),
+						d.Id(),
+					)
+				},
+			},
+		),
+		Importer: &schema.ResourceImporter{
+			StateContext: func(
+				ctx context.Context,
+				d *schema.ResourceData,
+				i interface{},
+			) ([]*schema.ResourceData, error) {
+				ids, err := unmarshalComposedID(d.Id(), "/", 2)
+				if err != nil {
+					return nil, fmt.Errorf(
+						"failed to unmarshal ID: %v",
+						err,
+					)
+				}
+				repositoryID := ids[0]
+				userAccountID := ids[1]
+				err = d.Set("repository_id", repositoryID)
+				if err != nil {
+					return nil, fmt.Errorf(
+						"failed to set 'repository_id': %v",
+						err,
+					)
+				}
+				d.SetId(userAccountID)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "ID of this resource in Cyral environment.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"repository_id": {
+				Description: "ID of the repository.",
+				Required:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+
+			"name": {
+				Description: "The name of the User Account.",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+
+			"auth_database_name": {
+				Description: "For User Accounts associated with a MongoDB cyral_repository, " +
+					"This value is only meaningful for User Accounts associated with " +
+					"MongoDB.",
+				Optional: true,
+				Type:     schema.TypeString,
+			},
+
+			"approval_config": {
+				Description: "Configurations related to Approvals.",
+				Optional:    true,
+				Type:        schema.TypeSet,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"automatic_grant": {
+							Description: "If true, it is possible to create an approval with an automatic grant" +
```

Comment:
```suggestion
							Description: "If `true`, it is possible to create an approval with an automatic grant" +
```

Same cosmetic markdown suggestion as before. :)